lib/error-handler.js

```diff
@@ -62,13 +63,31 @@ function handleError (reply, error, cb) {
     return
   }
 
-  const result = func(error, reply.request, reply)
-  if (result !== undefined) {
-    if (result !== null && typeof result.then === 'function') {
-      wrapThenable(result, reply)
-    } else {
-      reply.send(result)
+  try {
+    const result = func(error, reply.request, reply)
+    if (result !== undefined) {
+      if (result !== null && typeof result.then === 'function') {
+        wrapThenable(result, reply)
+      } else {
+        reply.send(result)
+      }
+    }
+  } catch (err) {
+    reply[kReplyIsError] = true
+    try {
+      if (err.cause) {
+        throw new Error('Child error already has a cause set')
+      }
+      err.cause = error
+    } catch (failedToAssignError) {
+      reply.log.warn({
+        childError: error,
+        parentError: err,
+        reason: failedToAssignError,
+        message: 'Failed to assign child error as cause to parent error thrown from error handler'
+      })
```

Comment:
You can avoid this try catch by putting the log line inside the if. I would avoid creating a new error too.

Reply:
This will break handling frozen / sealed objects. I can make this change if we're really sure we don't want to handle this: but feels risky to ignore edge case in error handler which if it throws the entire server crashes.

Reply:
I don't understand. How?

Reply:
Line 81 will throw if err is sealed or frozen, or is otherwise an odd object. This is separate to the if on line 78-80.

See test: https://github.com/fastify/fastify/pull/5222/files#diff-7cce5a5fdda1e1d2652f9fe0f081a7a7c580c090c9facd12ba36a65dddd4884eR168-R201